### DISCUSSION TITLE
Show only pinned tabs

### DISCUSSION THREAD

**Discussion by 3DRaven** [piroor/treestyletab#2816] [Q&A]
Hi! Cool addon :) How about option to show only pinned tabs in side panel? It can be help show many of pinned tabs in few columns. It is possible with this extension?

**Comment by 3DRaven** [comment]
Now, if I close all unpinned tabs, then I can't set width of side pannel less than 7 columns of pinned tabs. It is bad behavior for this functionality :) And I think best solution it show unpinned tabs in top (default for firefox) and show pinned tabs in side panel with what I need columns.

Tnx.

**Comment by 3DRaven** [comment]
Screenshot 

https://ibb.co/J7QJbTY

**Comment by 3DRaven** [comment]
And in this mode need to hide pinned tabs in default top panel of FF. This can be 

{code}
.tabbrowser-tab[pinned="true"] {
    display: none !important;
} 
{code}

**Comment by 3DRaven** [comment]
And need scroll of pinned tabs too.

**Comment by 3DRaven** [comment]
Now I used this css in userChrome.css

{code}
.tabbrowser-tab[pinned="true"] {
    display: none !important;
} 

#sidebar-box[sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"] #sidebar-header {
  display: none;
}

#sidebar {
  min-width: 10px !important;
}   
{code}

And get only pinned tabs panel without scroll. How to add scroll to the right side of side panel? And better is totaly hide unpinned tabs from this panel :)

**Comment by 3DRaven** [comment]
Screenshot

https://ibb.co/PYTj06M

**Comment by piroor** [comment]
I convert this to a discussion because this looks to be not an issue report.

**Comment by piroor** [comment]
You can hide unpinned tabs in the sidebar with a custom style rule (TST options => Advanced):

```css
tab-item:not(.pinned) {
  display: none;
}
```

Currently pinned tabs cannot become scrollable in the sidebar, but it should become available on the next release with the commit  8b91cc19a.

**Reply by 3DRaven** [reply]
Thank you! :) And another question, how to disable tab hierarchy functionality? I do not need to close all child tabs with parent and other.

**Reply by 3DRaven** [reply]
Current screenshot

https://ibb.co/3zNKh9t

**Reply by 3DRaven** [reply]
And how to hide "+" on the down side of panel?

**Reply by piroor** [reply]
> how to disable tab hierarchy functionality? I do not need to close all child tabs with parent and other.

There is a FAQ topic "I don't need automatically organized tree, instead I just want to organize tree by myself", please check it.
https://github.com/piroor/treestyletab#feature-requests
However, basically TST is designed to operate tabs with tree structure, so some built-in behaviors (e.g. closing of a collapsed tree closes all of descendants tabs also) are not customizable. Vertical tabs without hierarchy is truly out of its purpose, so I strongly recommend you to migrate from TST to something other vertical tab bar addon.


> And how to hide "+" on the down side of panel?

Did you try this code snippet?
https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#hide-the-new-tab-button-at-the-bottom-edge-of-the-tab-bar-1591
If there is no desired code snippet for your hope, you can inspect the DOM tree of TST's sidebar dynamically and write your custom CSS based on the inspection result.
https://github.com/piroor/treestyletab/wiki/How-to-inspect-tree-of-tabs#how-to-inspect-the-sidebar

**Reply by 3DRaven** [reply]
It is all that I wanted thank you very match :) Now it is just vertical tab with pinned tabs at the right side of my wide screen. It is perfect! Thank you for excellent add-on!

**Reply by 3DRaven** [reply]
Screenshot
https://ibb.co/Q68XXFP

And I tryed to find some other variants of addons for vertical bar with pin tabs, but another variants was ugly and do not customizable.

**Reply by piroor** [reply]
Hmm, but TST also looks not friendly to this usecase - actually you can use TST for the purpose but it requires many hacks (including userChrome.css and custom user styles for TST itself). I think that a new addon which simply list pinned tabs as icons will match to your case, and it looks to be a good motivation to develop such an addon by yourself if there is no existing alternative.

**Reply by 3DRaven** [reply]
Now, all my "I want" satisfied, except scroll, but scroll is "nice to have". So, I just will continue use TST with all css hacks that I wrote in userChrome.css :) Development of my self addon is lot of time. Thank you!

**Reply by 3DRaven** [reply]
userChrome.css
{code}
#tabbrowser-tabs{
  min-height: unset !important;
  padding-inline-start: 0px !important
}

.tabbrowser-tab[pinned="true"] {
    display: none !important;
} 


#sidebar-box[sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"] #sidebar-header {
  display: none;
}


#sidebar {
  min-width: 10px !important;
}   
{code}
TST style css
{code}
tab-item:not(.pinned) {
  display: none;
}

.newtab-button-box {
  display: none;
}
#tabbar {
  bottom: 0 !important; /* Eliminate dead space on bottom */
}
{code}
